app/models/assignment.rb

```diff
@@ -37,8 +38,11 @@ class Assignment < ApplicationRecord
 
   validate :uniqueness_of_slug_across_organization
 
+  validate :starter_code_repository_is_a_template_repository
```

Comment:
Could rename to `starter_code_repository_is_template` to shorten it, or even use `repo`.